### PR TITLE
OGM-1434 Test case for MongoDB: ObjectId passed as parameter to query

### PR DESCRIPTION
**Pull request by DavideD**
https://hibernate.atlassian.net/browse/OGM-1434

### CODE REVIEW

File: mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/id/objectid/Bar.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 import javax.persistence.OneToMany;
 
 import org.bson.types.ObjectId;
+import org.jboss.logging.processor.util.Objects;
```
Comment (gsmet) [inline mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/id/objectid/Bar.java]: I don't think this is the one you want to use :)

File: mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/id/objectid/DoorMan.java [hunk 2]
```diff
@@ -12,6 +12,7 @@
 import javax.persistence.Id;
 
 import org.bson.types.ObjectId;
+import org.jboss.logging.processor.util.Objects;
```
Comment (gsmet) [inline mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/id/objectid/DoorMan.java]: Same here.

File: mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/id/objectid/MusicGenre.java [hunk 3]
```diff
@@ -16,6 +16,7 @@
 import javax.persistence.OneToMany;
 
 import org.bson.types.ObjectId;
+import org.jboss.logging.processor.util.Objects;
```
Comment (gsmet) [inline mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/id/objectid/MusicGenre.java]: Ditto.

File: mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/id/objectid/Bar.java (hunk 1 above)
Comment (DavideD) [reply mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/id/objectid/Bar.java]: Damn! :)
Thanks

Comment (Sanne) [reply mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/id/objectid/Bar.java]: Add it to the checkstyle blacklist? We did similar things for e.g. to ban the deprecated junit helpers.

**Comment by DavideD** [comment]
Fixed the imports, Is it OK if I merge it @gsmet?

File: mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/id/objectid/Bar.java [hunk 4]
```diff
@@ -72,4 +73,31 @@ public void setMusicGenre(MusicGenre musicGenre) {
...
 		this.doorMen = doorMen;
 	}
+
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ( ( doorMen == null ) ? 0 : doorMen.hashCode() );
+		result = prime * result + ( ( musicGenre == null ) ? 0 : musicGenre.hashCode() );
+		result = prime * result + ( ( name == null ) ? 0 : name.hashCode() );
```
Comment (gsmet) [inline mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/id/objectid/Bar.java]: Note that in the tests, you can also use `Objects#hash(Object...)`. Not that I require you to change it here, just for next time.

It's not always recommended in our internal code as it allocates an array but for the tests, I think it's good enough.

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**